examples/acoustic/Acoustic_codegen.py

```diff
@@ -63,17 +63,16 @@ def damp_boundary(damp):
                               dtype=self.dtype, nbpml=nbpml)
         self.src.data[:] = data.get_source()[:, np.newaxis]
 
-        if auto_tuning:  # auto tuning with dummy forward operator
-            fw = ForwardOperator(self.model, self.src, self.damp, self.data,
-                                 time_order=self.t_order, spc_order=self.s_order,
-                                 save=False, profile=True)
-            self.at = AutoTuner(fw)
-            self.at.auto_tune_blocks(self.s_order + 1, self.s_order * 4 + 2)
-
     def Forward(self, save=False, cache_blocking=None,
                 auto_tuning=False, cse=True, compiler=None):
+
         if auto_tuning:
-            cache_blocking = self.at.block_size
+            fw = ForwardOperator(self.model, self.src, self.damp, self.data,
+                                 time_order=self.t_order, spc_order=self.s_order,
+                                 profile=True, save=False, cse=cse, compiler=compiler)
+            at = AutoTuner(fw)
+            at.auto_tune_blocks(self.s_order + 1, self.s_order * 4 + 2)
+            cache_blocking = at.block_size
 
```

Comment:
OK, I'm gonna now push a change that moves AT back into __init__. For this, ``cse`` and ``compiler`` are passed to ``Acoustic_cg``. 

Many parameters, however, are now passed to *both* the constructor (__init__) and the operators methods. Can we not just pass them to __init__, and then use ``self.`` in all methods? or do you want a default behaviour + per-operator knobs, as it would be now?